packages/table/src/table.tsx

```diff
@@ -681,28 +681,36 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
             selectionModes,
         } = nextProps;
 
-        const newChildArray = React.Children.toArray(children) as Array<React.ReactElement<IColumnProps>>;
+        const hasNewChildren = this.props.children !== nextProps.children;
+        const newChildArray = hasNewChildren ? React.Children.toArray(children) as Array<React.ReactElement<IColumnProps>> : this.childrenArray;
         const numCols = newChildArray.length;
 
-        // Try to maintain widths of columns by looking up the width of the
-        // column that had the same `ID` prop. If none is found, use the
-        // previous width at the same index.
-        const previousColumnWidths = newChildArray.map((child: React.ReactElement<IColumnProps>, index: number) => {
-            const mappedIndex = this.columnIdToIndex[child.props.id];
-            return this.state.columnWidths[mappedIndex != null ? mappedIndex : index];
-        });
-
-        // Make sure the width/height arrays have the correct length, but keep
-        // as many existing widths/heights when possible. Also, apply the
-        // sparse width/heights from props.
+        let gridInvalidationRequired = false;
         let newColumnWidths = this.state.columnWidths;
-        newColumnWidths = Utils.arrayOfLength(newColumnWidths, numCols, defaultColumnWidth);
-        newColumnWidths = Utils.assignSparseValues(newColumnWidths, previousColumnWidths);
-        newColumnWidths = Utils.assignSparseValues(newColumnWidths, columnWidths);
+        if (defaultColumnWidth !== this.props.defaultColumnWidth || columnWidths !== this.props.columnWidths || (hasNewChildren && !this.areChildrenSameShape(newChildArray))) {
```

Comment:
`columnWidths !== this.props.columnWidths` is problematic as array is likely to be a new instance?

Reply:
I guess my assumption here is that clients who are performance-sensitive will take care to memoize these arrays where possible, which is standard practice to get the benefits of pure-render semantics in react.